python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -2840,7 +2841,7 @@ def test_series_all_null(num_elements, null_type):
 @pytest.mark.parametrize("num_elements", [0, 2, 10, 100])
 def test_series_all_valid_nan(num_elements):
     data = [np.nan] * num_elements
-    sr = cudf.Series(data, nan_as_null=False)
+    sr = _create_cudf_series(data, nan_as_null=False)
```

Comment:
Why is this change needed?

For `num_elements == 0`, the null count should still be zero even if it interprets the dtype as string/object (so the test is still valid, though it might warn?).

For `num_elements > 0`, the data should be non-empty and thus be correctly interpreted as a floating type.